Ignore Redis firewall rules for caches using private endpoints

BenjaminEngeset opened the pull request:
## PR Summary

Fixes #2171 

Ignore Redis firewall rules for caches using private endpoints.

Updated `Azure.Redis.FirewallRuleCount` and `Azure.Redis.FirewallRuleIPRange`.

Added pre-condition and updated rule docs.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [ ] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

BenjaminEngeset commented:
Hi @BernieWhite you let me know.

BernieWhite commented on src/PSRule.Rules.Azure/rules/Azure.Redis.Rule.ps1:
```diff
@@ -181,13 +181,28 @@ function global:IsEnterpriseCache {
+        
+    )
+    process {
+        if ($PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled') {
+            $True
+        }
+        else {
+            $False
+        }
```
Can be simplified to `return $PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled';` however we also need to check the case when the `Microsoft.Cache/redis/firewallRules` type is used.

If `Microsoft.Cache/redis/firewallRules` is used, we should just check it as normal.

```suggestion
        return $PSRule.TargetType -eq 'Microsoft.Cache/redis/firewallRules' -or ($PSRule.TargetType -eq 'Microsoft.Cache/redis' -and $TargetObject.properties.publicNetworkAccess -ne 'Disabled');
```

BernieWhite reviewed (changes requested):
Nice work @BenjaminEngeset. Also thanks for the documentation updates. They look great. See comments below relating to `Microsoft.Cache/redis/firewallRules`.

Let's add an additional test to check for this case.

BenjaminEngeset commented:
Thanks @BernieWhite for identifying the bug I was close to implement 😿!

I've added the improved logic in the helper function and added a test case where the public network access is disabled.

BernieWhite reviewed (approved):
Thanks @BenjaminEngeset. The updates look good. All good to merge.
